src/pretix/plugins/paypal/views.py

```diff
@@ -76,40 +111,134 @@ def redirect_view(request, *args, **kwargs):
     return r
 
 
+# ToDo: cart-namespacing for widget?
+@method_decorator(csrf_exempt, name='dispatch')
+@method_decorator(xframe_options_exempt, 'dispatch')
+class XHRView(TemplateView):
+    template_name = ''
+
+    def post(self, request, *args, **kwargs):
+        if 'order' in self.kwargs:
+            order = self.request.event.orders.filter(code=self.kwargs['order']).select_related('event').first()
+            if order:
+                if order.secret.lower() == self.kwargs['secret'].lower():
+                    pass
+                else:
+                    order = None
```

Comment:
```suggestion
                if order.secret != self.kwargs['secret']:
                    order = None
```
I don't think these are case-insensitive in other places?

Reply:
This is verbatim from `/presale/views/order.py`: 
https://github.com/pretix/pretix/blob/19881dff443c5503afaaedc9b06b5504b6a4346e/src/pretix/presale/views/order.py#L102-L115

So I guess your choice how to proceed :-)